Code/client/ModCompat/BehaviorVar.cpp

```diff
@@ -0,0 +1,754 @@
+    // to their respective vectors
+
+    // The hashFile *__hash.txt file should contain the hashed signature of the ORIGINAL GAME
+    // actor. That enables us to look up the synced animation variables selected by 
+    // the STR devs and include them in the updated behavior/animation sync. Mod
+    // devs don't have to know anything but what they've added, and this support for
+    // Nemesis will work with more versions of the game.
+    //
+    // The signatureFile *__sig.txt should contain the name of a uniquely named variable for the
+    // modded actor, that we can use to connect the actor to their modded behavior variables.
+    // It's not actually sync'ed, just there so we can find the match. For example, currently
+    // bSTRMaster is used for humanoid/Player actors, and bSTRDragon for dragon behavior.
+    std::string hashFile;
+    std::string signatureFile;
+    std::vector<std::string> floatVarsFile;
+    std::vector<std::string> intVarsFile;
+    std::vector<std::string> boolVarsFile;
+    
+    for (auto& p : std::filesystem::directory_iterator(aDir))
+    {
+        std::string path = p.path().string();
+        std::string base_filename = path.substr(path.find_last_of("/\\") + 1);
+
+        spdlog::debug("base_path: {}", base_filename);
+
+        if (base_filename.find("__sig.txt") != std::string::npos)
+        {
+            signatureFile = path;
+
+            spdlog::debug("signature variable file: {}", signatureFile);
+        }
+        else if (base_filename.find("__hash.txt") != std::string::npos)
+        {
+            hashFile = path;
+
+            spdlog::debug("hash file: {}", hashFile);
+        }
+        else if (base_filename.find("__float.txt") != std::string::npos)
+        {
+            floatVarsFile.push_back(path);
+
+            spdlog::debug("float file: {}", path);
+        }
+        else if (base_filename.find("__int.txt") != std::string::npos)
+        {
+            intVarsFile.push_back(path);
+
+            spdlog::debug("int file: {}", path);
+        }
+        else if (base_filename.find("__bool.txt") != std::string::npos)
+        {
+            boolVarsFile.push_back(path);
+
+            spdlog::debug("bool file: {}", path);
+        }
+    }
+
+    // Check that there is a signature file (an identifying variable)
+    // When an Actor's behavior is modified, it's animation signature doesn't match
+    // any of the base STR built-in descriptors. We try to match up on a distinguishing
+    // animation variable added by the animation. But without such a distinquishing
+    if (signatureFile == "")
+        return nullptr;
+
+    // Prepare reading files
+    std::string sigVar;
+    std::string creatureName = aDir.filename().string();
+    std::vector<std::string> floatVar;
+    std::vector<std::string> intVar;
+    std::vector<std::string> boolVar;
+    std::string tempString;  // Temp string
+
+    std::ifstream fileSig(signatureFile);
```

Comment:
You cannot assume that this succceeds, handle error.

Reply:
Commit 2d989d98047dfa179045b7ba45333d5f17575a1b

Fixed it. It's a pretty narrow window, though. The path is from a directory iterator just run; so either the file was just deleted in the last millisecond or the directory is read-protected; a lot more would break in the latter case.